### DISCUSSION TITLE
M2M Improved API-Call

### DISCUSSION THREAD

**Discussion by JonathanSchndr** [directus/directus#15686] [Feature Requests]
Building on my issue (https://github.com/directus/directus/issues/15683) where I had a false expectation, I would like to put the following changes up for discussion:

Currently, an M2M works as follows:

I have a Collection Qualifcation and Collection Type. Qualification can have X many Types and Types can have X many Qualifcation (M2M). In the background the Junction Table is created automatically. If I now want to assign a Type to a new and existing Qualifcation (e.g. Types: High (Id: 10), Mid (Id: 11), Low (Id: 12)) this does not work with

`{"name":"Qualifcation A", type: [10,11]}`

but with 

`{"name":"Qualifcation A", "type":{"create":[ {"type_id":{"id":10}}, {"type_id":{"id":11}} ]}}`

Because in the first example tried the Junction ID is addressed and not the Qualification ID.  It returns an error (FORBIDDEN), because of course no junction may be created. Following I have the problem, if a user now deselects a type of his qualification (ID: 11). Do I first find out which Junction ID contains the relation and then delete it.

It would be easier if I could just pass in the Type ID in the array syntax and the API deletes the correct Junction entry for me. I think the API could be stripped down to the point where a junction table is no longer needed for the API consumer.

Also the GET command of a M2M returns the junction ID in the array, with an extension fields=type.* the correct type ID can be read out directly - but here I would like to ask again, why does the API consumer have to work with the Junctuon at all?

My suggestion is to use the junction table internally for processing of course. But not to let the API users work with it themselves. I think this is up to the API or are there areas where it is mandatory? Thanks for the discussion!

**Comment by br41nslug** [comment]
I am personally a fan of the explicitness of the current system although it does require that you know how the database is built up which might not always be as obvious in the App itself.
I'm not sure we can abstract the junction table away here as there are plenty of use-cases that require extra fields on the junction table to be accessible.
For instance:
- m2m/m2a have their `sort` field on the junction
- an Order has many Items with an `amount` on the junction

**Comment by rokdd** [comment]
Well it is kind of offtopic but I came around the same problem and I needed some minutes to figure out why I get a forbidden for my m2m relationship. Even with the documentation it is not clear that you have to use the junction ID (or field). What is also unclear, when I have a existing item and I want just to add the relationship to a m2m field - is this update or create when using the detailed mode?
I agree for both arguments of being "explicit" and "easy" are correct.. maybe a own route for the field might have a more simple logic like: `<item_id>/field_m2m/(append|delete|update)`

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
